src/vstart.sh

```diff
@@ -278,10 +282,14 @@ shift
 done
 
 if [ "$overwrite_conf" -eq 0 ]; then
-  CEPH_NUM_MON=`awk -F= '/CEPH_NUM_MON/{print $2}' $conf_fn`
-  CEPH_NUM_OSD=`awk -F= '/CEPH_NUM_OSD/{print $2}' $conf_fn`
-  CEPH_NUM_MDS=`awk -F= '/CEPH_NUM_MDS/{print $2}' $conf_fn`
-  CEPH_NUM_RGW=`awk -F= '/CEPH_NUM_RGW/{print $2}' $conf_fn`
+    MON=`awk -F= '/CEPH_NUM_MON/{print $2}' $conf_fn`
```

Comment:
if we go this way, we will always continue with whatever `CEPH_NUM_{MON,OSD,MDS,RGW}` the last vstart run used, even if "-k" is not passed in. in other words,
```bash
$ MON=1 OSD=1 MDS=0 ./vstart.sh
$ ./stop
$ MON=1 OSD=2 MDS=0 ./vstart.sh
# what? we have only 1 OSD?
```

if we move these lines after line 91, we overwrite the options for a non "-k" vstart cluster.